src/display/canvas.js

```diff
@@ -921,15 +929,7 @@ function getImageSmoothingEnabled(transform, interpolate) {
   const actualScale = Math.fround(
     (globalThis.devicePixelRatio || 1) * PixelsPerInch.PDF_TO_CSS_UNITS
   );
-  if (interpolate !== undefined) {
-    // If the value is explicitly set use it.
-    return interpolate;
-  } else if (scale[0] <= actualScale || scale[1] <= actualScale) {
-    // Smooth when downscaling.
-    return true;
-  }
-  // Don't smooth when upscaling.
-  return false;
+  return scale[0] <= actualScale && scale[1] <= actualScale;
```

Comment:
Shouldn't this be `||` instead of `&&` like before? I would think that if only one of the dimensions is reduced we would still need to apply some smoothing, or am I mistaken here?

Reply:
Good question, I asked me the same.
And I think that we've an upscale when only one dimension is upscaled and I think in this case we must respect the value of the flag.
That said in Firefox it's a &&:
https://searchfox.org/mozilla-central/rev/3c194fa1d6f339036d2ec9516bd310c6ad612859/dom/canvas/CanvasRenderingContext2D.cpp#4927-4928

Reply:
If one dimension is equal and the other one is downscaled it should still smoothen, right?

Reply:
Hm, I think you're right. That does make me wonder about the Firefox code though, and brings me back to the initial question of using `||` , so I guess I'm just not entirely sure what would be correct here. Considering that the reference tests didn't show many changes with the current version of the patch I'm not too worried about regressions here, so I'm also fine with keeping the current logic and adapting it later on if need be (because in that case we'll also have a test case that triggers the behavior).